tests/components/bond/test_config_flow.py

```diff
@@ -0,0 +1,115 @@
+
+from homeassistant import config_entries, setup
+from homeassistant.components.bond.const import DOMAIN
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_FRIENDLY_NAME, CONF_HOST
+
+from tests.async_mock import patch
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.bond.config_flow.Bond.getDeviceIds", return_value=[],
+    ), patch(
+        "homeassistant.components.bond.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.bond.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.1",
+                CONF_ACCESS_TOKEN: "test-token",
+                CONF_FRIENDLY_NAME: "test-friendly-name",
+            },
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "test-friendly-name"
+    assert result2["data"] == {
+        CONF_HOST: "1.1.1.1",
+        CONF_ACCESS_TOKEN: "test-token",
+        CONF_FRIENDLY_NAME: "test-friendly-name",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass):
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.bond.config_flow.Bond.getDeviceIds",
+        side_effect=JSONDecodeError("test-message", "test-doc", 0),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.1",
+                CONF_ACCESS_TOKEN: "test-token",
+                CONF_FRIENDLY_NAME: "test-friendly-name",
+            },
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.bond.config_flow.Bond.getDeviceIds",
+        side_effect=ConnectionError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.1",
+                CONF_ACCESS_TOKEN: "test-token",
+                CONF_FRIENDLY_NAME: "test-friendly-name",
```

Comment:
im not sure id worry about friendly names cause names can be changed in UI since a unique_id exists. Does the hub offer its name via API?

Reply:
No, the hub does not provide any name over its API. I also made the friendly name optional. I am happy to remove it completely if this is considered best practice. LMK. Thanks!

Reply:
I would say remove it because a user can rename the integration from UI in recent versions as well if they didnt like the hostname. You may also be able to get a name during discovery or API (future) which would mean this would end up removed/deprecated in favor of those values

Reply:
thanks, removed